### ISSUE TITLE
Efficient broadcast comparison

### ISSUE THREAD

**Issue by cstjean** [JuliaData/CategoricalArrays.jl#150]
On 0.6,

```julia
julia> heys1 = fill("Hey", 10000);

julia> heys3 = CategoricalVector(heys1);

julia> @btime heys1 .== "Hey";
  62.509 μs (21 allocations: 6.19 KiB)

julia> @btime heys3 .== "Hey";
  101.900 μs (21 allocations: 6.19 KiB)
```

Theoretically the second comparison should be faster, since it should boil down to comparing integers (or, in this case, realizing that no value in the pool is `==`). Is there a way of implementing this? I'm not familiar with broadcasting innards.

**Comment by nalimilan** [comment]
Yes, that would be a nice optimization. I'm not sure how hard it would be to override the `broadcast` machinery to implement this. One difficulty is that only the case where the second argument is a scalar can be handled efficiently, for more complex situations the generic fallback is OK.
